CMake/CheckCompiler.cmake

```diff
@@ -0,0 +1,15 @@
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```

Comment:
Actually not too sure which compilers we should limit to. My best guess is pick the latest compilers that come with Ubuntu and MacOSX these days and set those as minimum requirement.